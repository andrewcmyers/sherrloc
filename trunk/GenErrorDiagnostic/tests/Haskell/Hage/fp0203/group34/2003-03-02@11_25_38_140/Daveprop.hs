module Daveprop where
eqString      :: String -> String -> Bool 
eqString = undefined
data Prop
  = En [Prop]
  | Of [Prop]
  | Niet Prop
  | Prop :-> Prop
  | Var String
  | Bool Bool

type Bedeling = [String]

evalueer :: Prop -> Bedeling -> Bool
evalueer (Var vartje) [] = False
evalueer (Var vartje) (x:xs)  | eqString vartje x = True
                              | otherwise             = evalueer vartje xs
evalueer (En (x:xs)) bedje = evalueer x bedje && evalueer (En (xs)) bedje
evalueer (Of (x:xs)) bedje = evalueer x bedje || evalueer (Of (xs)) bedje

-- vartje should be: (Var vartje)
-- 17,66-71
