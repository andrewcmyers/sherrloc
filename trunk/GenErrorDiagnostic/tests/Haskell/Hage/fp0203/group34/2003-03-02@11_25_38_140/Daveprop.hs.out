Compiling ./group34/2003-03-02@11:25:38:140/Daveprop.hs
(12,15): Warning: Variable "vartje" is not used
(14,66): Type error in application
 expression       : evalueer vartje xs
 function         : evalueer
   type           : Prop -> Bedeling -> Bool
 1st argument     : vartje
   type           : String
   does not match : Prop  

Compilation failed with 1 error
